clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -271,53 +246,70 @@ struct BuiltinTypeDeclBuilder {
     return *this;
   }
 
+  FieldDecl *getResourceHandleField() {
+    FieldDecl *FD = Fields["h"];
```

Comment:
Won't this add "h" when accessed this way?
How about:
```suggestion
    FieldDecl *FD = Fields.lookup("h");
```

Reply:
Both [] operator and lookup will add the field if it is not already there. I will update it to use "find".

Reply:
I thought `lookup` was right:
https://github.com/llvm/llvm-project/blob/30af6fb163add17a6be515200881afdff91d213a/llvm/include/llvm/ADT/StringMap.h#L251-L258

Reply:
Yeah, it would probably work fine too, if default value `ValueTy()` just means `nullptr` when `ValueTy` is a pointer.
I looked at the `lookup` documentation and it says: "Return the entry for the specified key, or a default constructed value if no such entry exists."